odoo/addons/base/controllers/avatar.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+
+from odoo.http import Controller, request, route, Response
+
+# show avatar if no image or showAvatar 1
+# show image if image
+
+class Avatar(Controller):
+    @route(['/web/avatar/<int:user_id>', '/web/avatar/<int:user_id>/<string:image_field>'], methods=['GET'])
+    def get_avatar(self, user_id=1, image_field="image_128"):
+        user = request.env['res.users'].browse(user_id).exists()
+        if user:
+            avatar = _generate_avatar_svg(user.name, user.create_date)
+            return Response(avatar, mimetype='image/svg+xml')
+        return Response("User not found.", status=500)
+
+def _generate_avatar_svg(name, seed=False):
+    initials = _generate_initials(name)
+    bgcolor = _generate_bgcolor(f"{name}{seed.timestamp()}")
+    if initials and bgcolor:
+        return (
+            "<?xml version='1.0' encoding='UTF-8' ?>"
+            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+            f"<rect fill='{bgcolor}' height='180' width='180'/>"
+            f"<text fill='#ffffff' font-size='64' text-anchor='middle' x='90' y='106' font-family='sans-serif'>{initials}</text>"
+            "</svg>"
+        )
```

Comment:
When hand-crafting XML, it is *always* required to escape *any* data that your inject into the code, even if it looks very safe at the moment. More info [here](https://github.com/odoo/enterprise/pull/15297#issuecomment-768985663).
I would usually recommend a qweb template where `t-esc` does the job for you, but we've established that it was overkill here. So escaping you must!

```suggestion
    initials = tools.html_escape(_generate_initials(name))
    bgcolor = tools.html_escape(_generate_bgcolor(f"{name}{seed.timestamp()}"))
    if initials and bgcolor:
        return (
            "<?xml version='1.0' encoding='UTF-8' ?>"
            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
            f"<rect fill='{bgcolor}' height='180' width='180'/>"
            f"<text fill='#ffffff' font-size='64' text-anchor='middle' x='90' y='106' font-family='sans-serif'>{initials}</text>"
            "</svg>"
        )
```

(watch out for the missing import for `odoo.tools`)

Reply:
>  I would usually recommend a qweb template where `t-esc` does the job for you, but we've established that it was overkill here. So escaping you must!

Other nice alternatives to a qweb template:
* Use `etree`, possibly the ElementBuilder system, then convert that to a string.
* Use `markupsafe.Markup`, it behaves like a string but automatically escapes what you inject (doesn't work with f-strings but it works with both `%` and `.format`).

*Technically* markupsafe is not a direct dependency but it's a mandatory indirect one (through jinja) so it wouldn't be much of a stretch to add it to the requirements file, it doesn't actually change anything to our dependency set.